### PR TITLE
Splits probes for mongodb between server and database

### PR DESCRIPTION
**Pull request by solatis**
This PR deals with two issues:

 * mongodb's `node_status` probe has been split into `server_probe` and `db_probe`;
 * the UI is now (a little bit) more flexible with which charts it displays.

### CODE REVIEW

File: html_report/bench/series/detail.js [hunk 1]
```diff
@@ -34,7 +34,23 @@ function makeLineFunction(serie, op) {
     };
 }
 
-var lineSeries = [
+function lineSeriesFromTest(test) {
+    var series = defaultLineSeries;
```
Comment (bblanchon) [inline html_report/bench/series/detail.js]: This is a reference, not a copy

File: html_report/bench/series/detail.js [hunk 2]
```diff
@@ -34,7 +34,23 @@ function makeLineFunction(serie, op) {
...
+
+    for (var key in test) {
+        if (optionalLineSeries[key]) {
+            series.push(optionalLineSeries[key]);
```
Comment (bblanchon) [inline html_report/bench/series/detail.js]: You're altering the original "defaultLineSeries"

File: html_report/main.js [hunk 3]
```diff
@@ -6,6 +6,8 @@ function addTestResult(test) {
     test.id = test_by_id.length;
     test_by_id.push(test);
 
+    lineSeries = lineSeriesFromTest(test);
```
Comment (bblanchon) [inline html_report/main.js]: The variable lineSeries isn't used.

**Comment by bblanchon** [comment]
There is a problem with the way you generate the line series.
It works if all you test contains the same traces, but it's probably not what you intended to write.

File: html_report/bench/series/detail.js (hunk 1 above)
Comment (solatis) [inline html_report/bench/series/detail.js]: Good point! Will fix this.

File: html_report/main.js (hunk 3 above)
Comment (solatis) [inline html_report/main.js]: lineSeries is a global variable defined in serials/detail.js, which is assigned here.

**Comment by solatis** [comment]
It's a quick hack I wrote in 30 minutes to be able to get meaningful benchmarks out of MongoDB. I think the best way to fix this is what we discussed earlier, that the different series are actually defined ad-hoc in the benchmark, not in javascript. 

This is actually not at all that difficult if we let each c++ probe define information about the series it emits, and emit the appropriate javascript code in the results.jsonp file. Shall I implement this while you focus on the SNMP code?

**Comment by bblanchon** [comment]
I merged the 2 first commits.
I didn't merge the last commit because it doesn't work when you have several kinds of tests in your results (`std_*`  and `qdb_*` for instance).
Don't worry, I'll rework tht part soon, while adding the SNMP probe.

**Pull request closed without merging**